dpnp/dpnp_iface_linearalgebra.py

```diff
@@ -97,11 +97,18 @@ def dot(x1, x2, **kwargs):
         dim1 = x1_desc.ndim
         dim2 = x2_desc.ndim
 
-        if not (dim1 >= 2 and dim2 == 1) and not (dim1 >= 2 and dim2 >= 2) and (x1_desc.dtype == x2_desc.dtype):
+        # for now we work only with these cases
+        if (
```

Comment:
We could split the condition with code like below:
```python
if (dim1 == 1 and dim2 == 1) or (dim1 == 2 and dim2 == 2):
    pass
elif x1_desc.dtype != x2_desc.dtype:
    pass
else:
    result_obj = dpnp_dot(x1_desc, x2_desc).get_pyobj()
    if (dim1 == 2 and dim2 == 2):
        return result_obj
    else:
        result = dpnp.convert_single_elem_array_to_scalar(result_obj)
        return result
```

Reply:
> We could split the condition with code like below:
> 
> ```python
> if (dim1 == 1 and dim2 == 1) or (dim1 == 2 and dim2 == 2):
>     pass
> elif x1_desc.dtype != x2_desc.dtype:
>     pass
> else:
>     result_obj = dpnp_dot(x1_desc, x2_desc).get_pyobj()
>     if (dim1 == 2 and dim2 == 2):
>         return result_obj
>     else:
>         result = dpnp.convert_single_elem_array_to_scalar(result_obj)
>         return result
> ```

You want to return behavior back and send 2D and 1D cases in fallback :))
> ```if (dim1 == 1 and dim2 == 1) or (dim1 == 2 and dim2 == 2):
>     pass```

I think it is a typo, and I understand how to rewrite these new conditions in few lines, but I suggest leave it as is and with existing comments. This code should be changed one more time in nearest future because support of strides in matrix multiplication is requested.